test/e2e/framework/pod/wait.go

```diff
@@ -809,6 +809,18 @@ func WaitForPodContainerStarted(ctx context.Context, c clientset.Interface, name
 	})
 }
 
+// WaitForPodInitContainerStarted waits for the given Pod init container to start.
+func WaitForPodInitContainerStarted(ctx context.Context, c clientset.Interface, namespace, podName string, initContainerIndex int, timeout time.Duration) error {
+	conditionDesc := fmt.Sprintf("init container %d started", initContainerIndex)
+	return WaitForPodCondition(ctx, c, namespace, podName, conditionDesc, timeout, func(pod *v1.Pod) (bool, error) {
+		if initContainerIndex > len(pod.Status.InitContainerStatuses)-1 {
```

Comment:
```suggestion
		if initContainerIndex >= len(pod.Status.InitContainerStatuses) {
```

Reply:
/approve

This is is minor, but IMHO worth simplifying. I'll let you decide when you want to do that.

/hold

Reply:
it's a copy from `WaitForPodContainerStarted`. Let's fix them both later

/unhold